src/applications/edu-benefits/tests/complaint-tool/actions/schoolSearch.unit.spec.js

```diff
@@ -0,0 +1,89 @@
+import { expect } from 'chai';
+import sinon from 'sinon';
+import { mockFetch, resetFetch } from '../../../../../platform/testing/unit/helpers.js';
+import {
+  setCannotFindSchool,
+  searchInputChange,
+  searchSchools,
+  selectInstitution
+} from '../../../complaint-tool/actions/schoolSearch';
+
+function setFetchResponse(stub, data) {
+  const response = new Response();
+  response.ok = true;
+  response.json = () => Promise.resolve(data);
+  stub.resolves(response);
+}
+
+describe.only('schoolSearch actions', () => {
+  describe('setCannotFindSchool', () => {
+    it('should return a SET_DATA action', () => {
+      expect(setCannotFindSchool()).to.eql({
+        type: 'SET_DATA',
+        data: {
+          school: {
+            'view:cannotFindSchool': true
+          }
+        }
+      });
+    });
+  });
+  describe('searchInputChange', () => {
+    it('should return a SEARCH_INPUT_CHANGED action', () => {
+      expect(searchInputChange({ searchInputValue: 'test' })).to.eql({
+        type: 'SEARCH_INPUT_CHANGED',
+        searchInputValue: 'test'
+      });
+    });
+  });
+  describe('searchSchools', () => {
+    it('should dispatch LOAD_SCHOOLS_STARTED and LOAD_SCHOOLS_SUCCEEDED actions', (done) => {
+      const payload = { test: 'test' };
+      mockFetch();
+      setFetchResponse(global.fetch.onFirstCall(), payload);
+
+      const dispatch = sinon.spy();
+
+      const schoolSearchQuery = {
+        institutionQuery: 'testQuery',
+        page: 1
+      };
+
+      searchSchools(schoolSearchQuery)(dispatch);
```

Comment:
I'm sure we have some helpers for this but I usually prefer to just create and pass my own dispatch. I always have to go through a recall process when remembering how to use thunks and this kind of thing makes it easier for me.

Reply:
That makes sense.